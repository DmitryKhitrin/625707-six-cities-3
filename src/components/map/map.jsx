import React, {PureComponent} from "react";
import {PropTypes} from "prop-types";
import leaflet from "leaflet";

export class Map extends PureComponent {
  constructor(props) {
    super(props);
  }

  _initCities() {
    const {placeCardsList, city} = this.props;

    const icon = leaflet.icon({
      iconUrl: `img/pin.svg`,
      iconSize: [30, 30]
    });

    const zoom = 12;
    const map = leaflet.map(`map`, {
      center: city,
      zoom,
      zoomControl: false,
      marker: true
    });
    map.setView(city, zoom);
    leaflet
      .tileLayer(
          `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
          {
            attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
          }
      )
      .addTo(map);

    placeCardsList.map((card) => {
      leaflet.marker(card.coords, {icon, title: card.cardName}).addTo(map);
    });
  }

  componentDidMount() {
    this._initCities();
  }

  render() {
    return <div id="map" style={{height: 1000}} />;
  }
}

Map.propTypes = {
  city: PropTypes.arrayOf(PropTypes.number).isRequired,
  placeCardsList: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        priceValue: PropTypes.number,
        placeCardImage: PropTypes.string,
        cardName: PropTypes.string,
        starsRating: PropTypes.string,
        roomType: PropTypes.string,
        isPremium: PropTypes.bool,
        coords: PropTypes.arrayOf(PropTypes.number).isRequired
      })
  )
};

