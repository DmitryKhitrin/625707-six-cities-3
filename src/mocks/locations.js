export const locations = [
  `Paris`,
  `Dusseldorf`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
];
