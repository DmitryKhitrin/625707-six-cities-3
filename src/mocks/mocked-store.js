import configureStore from "redux-mock-store";

const mockStore = configureStore([]);
export const store = mockStore({
  offer: {
    locations: [],
    city: `Paris`,
    offers: []
  },
  user: {
    authorizationStatus: `NO_AUTH`,
    user: {
      id: null,
      email: ``,
      name: ``,
      avatar: ``,
      isPro: false
    },
    error: ``
  }
});
