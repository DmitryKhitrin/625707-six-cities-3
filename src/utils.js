import {sortTypes} from "./sortTypes.js";

const parseRating = (rate) => (String(Math.abs(rate * 100 / 5)) + `%`);

const parseUser = ({avatar_url: hostPhoto, name: hostName, id, is_prop: isProp}) => ({
  hostPhoto,
  hostName,
  id,
  isProp,
});

export const parseOffer = ({
  preview_image: previewImage,
  images,
  title,
  is_favorite: isFavorite,
  is_premium: isPremium,
  rating,
  type,
  bedrooms,
  max_adults: maxAdults,
  price,
  goods,
  host,
  location,
  description,
  id,
  city
}) => {
  return {
    previewImage,
    images,
    title,
    isFavorite,
    isPremium,
    rating: parseRating(rating),
    type,
    bedrooms,
    maxAdults,
    price,
    goods,
    host: parseUser(host),
    location: [location.latitude, location.longitude],
    description,
    id: String(id),
    city: city.name
  };
};

export const parseCities = (offers) => {
  const citiesList = new Set();
  return offers.map((offer) => {
    const {city} = offer;
    const {name, location} = city;
    return {
      name,
      location: [parseFloat(location.latitude), parseFloat(location.longitude)],
    };
  }
  ).filter((city) => {
    if (!citiesList.has(city.name)) {
      citiesList.add(city.name);
      return true;
    }
    return false;
  });
};

export const parseFavorites = (favorites) => {
  if (favorites.length === 0) {
    return {};
  }
  return favorites.reduce((acc, offer) => {
    const parsedOffer = parseOffer(offer);
    const {city} = parsedOffer;
    if (acc[city]) {
      acc[city] = [...acc[city], parsedOffer];
    } else {
      acc[city] = [parsedOffer];
    }
    return acc;
  }, {});
};

const rateToNumber = (rate) => (Number(rate.replace(`%`, ``)));

export const sortOffers = (type, offersList) => {
  switch (type) {
    case sortTypes.LOW_TO_HIGHT:
      return [...offersList].sort((a, b) => a.price - b.price);
    case sortTypes.HIGHT_TO_LOW:
      return [...offersList].sort((a, b) => b.price - a.price);
    case sortTypes.TOP_RATED:
      return [...offersList].sort(
          (a, b) => rateToNumber(b.rating) - rateToNumber(a.rating)
      );
    default:
      return [...offersList];
  }
};

export const parseComment = ({
  comment,
  date,
  id,
  rating,
  user,
}) => ({
  comment,
  date,
  id: String(id),
  rating: parseRating(rating),
  user: parseUser(user)
});

