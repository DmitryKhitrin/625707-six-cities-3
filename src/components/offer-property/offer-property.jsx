import React from "react";
import PropTypes from "prop-types";
import {ReviewsList} from "../reviews-list/reviews-list.jsx";
import {PropertiesInsideList} from "../properties-inside-list/properties-inside-list.jsx";
import {OffersList} from "../offers-list/offers-list.jsx";
import {Map} from "../map/map.jsx";

export const OfferPropperty = ({
  // placePhotosList = [],
  offerHeader = ``,
  // descriptions = ``,
  isPremium,
  placeType = ``,
  rating = ``,
  bedroomsCount = ``,
  maxPeopleCount = 4,
  price = 120,
  amenitiesList = [],
  reviews,
  offersList,
  // hostInformation
}) => {
  const premium = isPremium ? (
    <div className="property__mark">
      <span>Premium</span>
    </div>
  ) : null;
  return (
    <div>
      <meta charSet="utf-8" />
      <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <title>6 cities: property</title>
      <link rel="stylesheet" href="css/main.css" />
      <div style={{display: `none`}}>
        <svg xmlns="http://www.w3.org/2000/svg">
          <symbol id="icon-arrow-select" viewBox="0 0 7 4">
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
            />
          </symbol>
          <symbol id="icon-bookmark" viewBox="0 0 17 18">
            <path d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z" />
          </symbol>
          <symbol id="icon-star" viewBox="0 0 13 12">
            <path
              fillRule="evenodd"
              clipRule="evenodd"
              d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
            />
          </symbol>
        </svg>
      </div>
      <div className="page">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <a className="header__logo-link" href="main.html">
                  <img
                    className="header__logo"
                    src="img/logo.svg"
                    alt="6 cities logo"
                    width={81}
                    height={41}
                  />
                </a>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <a
                      className="header__nav-link header__nav-link--profile"
                      href="#"
                    >
                      <div className="header__avatar-wrapper user__avatar-wrapper"></div>
                      <span className="header__user-name user__name">
                        Oliver.conner@gmail.com
                      </span>
                    </a>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>
        <main className="page__main page__main--property">
          <section className="property">
            <div className="property__gallery-container container">
              <div className="property__gallery">
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/room.jpg"
                    alt="Photo studio"
                  />
                </div>
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/apartment-01.jpg"
                    alt="Photo studio"
                  />
                </div>
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/apartment-02.jpg"
                    alt="Photo studio"
                  />
                </div>
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/apartment-03.jpg"
                    alt="Photo studio"
                  />
                </div>
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/studio-01.jpg"
                    alt="Photo studio"
                  />
                </div>
                <div className="property__image-wrapper">
                  <img
                    className="property__image"
                    src="img/apartment-01.jpg"
                    alt="Photo studio"
                  />
                </div>
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                {premium}
                <div className="property__name-wrapper">
                  <h1 className="property__name">{offerHeader}</h1>
                  <button
                    className="property__bookmark-button button"
                    type="button"
                  >
                    <svg
                      className="property__bookmark-icon"
                      width={31}
                      height={33}
                    >
                      <use xlinkHref="#icon-bookmark" />
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={{width: rating}} />
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">
                    4.8
                  </span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {placeType}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {bedroomsCount}
                  </li>
                  <li className="property__feature property__feature--adults">
                    Max {maxPeopleCount} adults
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">â‚¬{price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <PropertiesInsideList propertyInside={amenitiesList} />
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                      <img
                        className="property__avatar user__avatar"
                        src="img/avatar-angelina.jpg"
                        width={74}
                        height={74}
                        alt="Host avatar"
                      />
                    </div>
                    <span className="property__user-name">Angelina</span>
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      A quiet cozy and picturesque that hides behind a a river
                      by the unique lightness of Amsterdam. The building is
                      green and from 18th century.
                    </p>
                    <p className="property__text">
                      An independent House, strategically located between
                      Rembrand Square and National Opera, but where the bustle
                      of the city comes to rest in this alley flowery and
                      colorful.
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <h2 className="reviews__title">
                    Reviews &middot;
                    <span className="reviews__amount">{reviews.length}</span>
                  </h2>
                  <ReviewsList reviews={reviews} />
                  <form className="reviews__form form" action="#" method="post">
                    <label
                      className="reviews__label form__label"
                      htmlFor="review"
                    >
                      Your review
                    </label>
                    <div className="reviews__rating-form form__rating">
                      <input
                        className="form__rating-input visually-hidden"
                        name="rating"
                        defaultValue={5}
                        id="5-stars"
                        type="radio"
                      />
                      <label
                        htmlFor="5-stars"
                        className="reviews__rating-label form__rating-label"
                        title="perfect"
                      >
                        <svg
                          className="form__star-image"
                          width={37}
                          height={33}
                        >
                          <use xlinkHref="#icon-star" />
                        </svg>
                      </label>
                      <input
                        className="form__rating-input visually-hidden"
                        name="rating"
                        defaultValue={4}
                        id="4-stars"
                        type="radio"
                      />
                      <label
                        htmlFor="4-stars"
                        className="reviews__rating-label form__rating-label"
                        title="good"
                      >
                        <svg
                          className="form__star-image"
                          width={37}
                          height={33}
                        >
                          <use xlinkHref="#icon-star" />
                        </svg>
                      </label>
                      <input
                        className="form__rating-input visually-hidden"
                        name="rating"
                        defaultValue={3}
                        id="3-stars"
                        type="radio"
                      />
                      <label
                        htmlFor="3-stars"
                        className="reviews__rating-label form__rating-label"
                        title="not bad"
                      >
                        <svg
                          className="form__star-image"
                          width={37}
                          height={33}
                        >
                          <use xlinkHref="#icon-star" />
                        </svg>
                      </label>
                      <input
                        className="form__rating-input visually-hidden"
                        name="rating"
                        defaultValue={2}
                        id="2-stars"
                        type="radio"
                      />
                      <label
                        htmlFor="2-stars"
                        className="reviews__rating-label form__rating-label"
                        title="badly"
                      >
                        <svg
                          className="form__star-image"
                          width={37}
                          height={33}
                        >
                          <use xlinkHref="#icon-star" />
                        </svg>
                      </label>
                      <input
                        className="form__rating-input visually-hidden"
                        name="rating"
                        defaultValue={1}
                        id="1-star"
                        type="radio"
                      />
                      <label
                        htmlFor="1-star"
                        className="reviews__rating-label form__rating-label"
                        title="terribly"
                      >
                        <svg
                          className="form__star-image"
                          width={37}
                          height={33}
                        >
                          <use xlinkHref="#icon-star" />
                        </svg>
                      </label>
                    </div>
                    <textarea
                      className="reviews__textarea form__textarea"
                      id="review"
                      name="review"
                      placeholder="Tell how was your stay, what you like and what can be improved"
                      defaultValue={``}
                    />
                    <div className="reviews__button-wrapper">
                      <p className="reviews__help">
                        To submit review please make sure to set{` `}
                        <span className="reviews__star">rating</span> and
                        describe your stay with at least{` `}
                        <b className="reviews__text-amount">50 characters</b>.
                      </p>
                      <button
                        className="reviews__submit form__submit button"
                        type="submit"
                        disabled
                      >
                        Submit
                      </button>
                    </div>
                  </form>
                </section>
              </div>
            </div>
            <section className="property__map map">
              <Map
                city={[52.38333, 4.9]}
                placeCardsList={offersList}
                height={600}
              />
            </section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">
                Other places in the neighbourhood
              </h2>
              <div className="near-places__list places__list">
                <OffersList
                  placeCardsList={offersList}
                  onHeaderClick={() => {}}
                  onMouseEnter={jest.fn()}
                  onMouseLeave={jest.fn()}
                />
              </div>
            </section>
          </div>
        </main>
      </div>
    </div>
  );
};

OfferPropperty.propTypes = {
  reviews: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string.isRequired,
        reviewsDate: PropTypes.string.isRequired,
        reviewsText: PropTypes.string.isRequired,
        reviewsRating: PropTypes.string.isRequired,
        reviewsUserName: PropTypes.string.isRequired,
        reviewsAvatar: PropTypes.string.isRequired
      })
  ),
  placePhotosList: PropTypes.arrayOf(PropTypes.string),
  offerHeader: PropTypes.string.isRequired,
  descriptions: PropTypes.arrayOf(PropTypes.string.isRequired),
  isPremium: PropTypes.bool.isRequired,
  placeType: PropTypes.string.isRequired,
  rating: PropTypes.string.isRequired,
  bedroomsCount: PropTypes.string.isRequired,
  maxPeopleCount: PropTypes.number.isRequired,
  price: PropTypes.number.isRequired,
  amenitiesList: PropTypes.arrayOf(PropTypes.string),
  hostInformation: PropTypes.shape({
    hostPhoto: PropTypes.string,
    hostName: PropTypes.string,
    isSuper: PropTypes.bool
  }),
  offersList: PropTypes.array.isRequired,
};
