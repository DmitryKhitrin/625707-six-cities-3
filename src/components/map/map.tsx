import React, {PureComponent} from "react";
import leaflet from "leaflet";
import {ParsedOfferCard, ParsedCity} from "../../utils/utils";


type Props = {
  placeCardsList: Pick<ParsedOfferCard, |`id` | `title` | `location`>[];
  city: ParsedCity;
  height: number;
  activeCard: string;
}

const SETTINGS = {
  zoom: 13,
};

export class Map extends PureComponent<Props> {

  _map: leaflet.Map = {} as leaflet.Map;
  _zoom = 13;
  _markersLayer: leaflet.LayerGroup<any> = {} as leaflet.LayerGroup<any>;
  _activeCard = ``;

  _getIcon() {
    return leaflet.icon({
      iconUrl: `img/pin.svg`,
      iconSize: [30, 30],
    });
  }

  _getActiveIcon() {
    return leaflet.icon({
      iconUrl: `img/pin-active.svg`,
      iconSize: [30, 30],
    });
  }

  _initCities() {
    const {city} = this.props;
    const {location = [52.38333, 4.9]} = city;
    this._map = leaflet.map(`map`, {
      center: location,
      zoom: SETTINGS.zoom,
      zoomControl: false,
    });
    this._map.setView(location, this._zoom);
    leaflet
             .tileLayer(
                 `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
                 {
                   attribution: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`
                 }
             )
             .addTo(this._map);

    this._addMarkers();
  }

  _addMarkers() {
    const {placeCardsList, activeCard} = this.props;
    this._activeCard = activeCard;
    this._markersLayer = leaflet.layerGroup().addTo(this._map);

    placeCardsList.forEach((card) => {
      const {location, title, id} = card;
      if (!location) {
        return;
      }
      const icon =
        this._activeCard === id ? this._getActiveIcon() : this._getIcon();
      leaflet
        .marker(location, {icon, title})
        .addTo(this._markersLayer);
    }

    );
  }

  componentDidMount() {
    this._initCities();
  }

  componentDidUpdate() {
    const {name, location} = this.props.city;
    if (name && location.length) {
      this._markersLayer.clearLayers();
      this._map.setView(location, this._zoom);
      this._addMarkers();
    }

  }

  render() {
    return <div id="map" style={{height: this.props.height}} />;
  }
}
