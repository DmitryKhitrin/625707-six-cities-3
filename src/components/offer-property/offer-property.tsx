import React, {FC, useCallback, memo} from 'react';
import {useHistory} from "react-router-dom";
import {ReviewsList} from "../reviews-list/reviews-list";
import {PropertiesInsideList} from "../properties-inside-list/properties-inside-list";
import {OffersList} from "../offers-list/offers-list";
import {Map} from "../map/map";
import {FeedbackFrom} from "../feedback-form/feedback-form";
import {ParsedOfferCard, ParsedComment} from "../../utils/utils";

type Props = ParsedOfferCard & {
  isAuthenticated: boolean;
  setFavorite: (T: string, S: number) => void;
  reviews: ParsedComment[];
  offersList: ParsedOfferCard[];
}

const OfferPropperty: FC<Props> = ({
  images = [],
  title,
  description = ``,
  isPremium,
  type = ``,
  rating = ``,
  bedrooms = ``,
  maxAdults = 4,
  price = 120,
  goods = [],
  reviews = [],
  offersList = [],
  isAuthenticated = false,
  host = {},
  city = {},
  isFavorite,
  setFavorite,
  id,
}) => {
  const {personPhoto = ``, personName = ``} = host;
  const {name, location} = city;
  const history = useHistory();

  const onFavoriteClick = useCallback(
      () => (isAuthenticated ? setFavorite(id, Number(!isFavorite)) : history.push(`/login`)),
      [setFavorite, isAuthenticated, id, isFavorite, history],
  );
  const premium = isPremium ? (
    <div className="property__mark">
      <span>Premium</span>
    </div>
  ) : null;
  return (
    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {images.map((item, index) => {
              return (
                <div key={index} className="property__image-wrapper">
                  <img
                    className="property__image"
                    src={item}
                    alt="Photo studio"
                  />
                </div>
              );
            })}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            {premium}
            <div className="property__name-wrapper">
              <h1 className="property__name">{title}</h1>
              <button
                className="property__bookmark-button button"
                type="button"
                onClick={onFavoriteClick}
              >
                <svg
                  className="property__bookmark-icon"
                  width={31}
                  height={33}
                >
                  <use xlinkHref="#icon-bookmark" />
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: rating}} />
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">4.8</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {type}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {bedrooms}
              </li>
              <li className="property__feature property__feature--adults">
                                      Max {maxAdults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">â‚¬{price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <PropertiesInsideList propertyInside={goods} />
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img
                    className="property__avatar user__avatar"
                    src={personPhoto}
                    width={74}
                    height={74}
                    alt="Host avatar"
                  />
                </div>
                <span className="property__user-name">{personName}</span>
              </div>
              <div className="property__description">
                <p className="property__text">{description}</p>
              </div>
            </div>
            <section className="property__reviews reviews">
              <h2 className="reviews__title">
                                      Reviews &middot;
                <span className="reviews__amount">{reviews.length}</span>
              </h2>
              <ReviewsList reviews={reviews} />
              {isAuthenticated ? <FeedbackFrom id={id} /> : null}
            </section>
          </div>
        </div>
        <section className="property__map map">
          <Map
            city={{location, name}}
            placeCardsList={[...offersList, {id, location, title}]}
            activeCard={id}
            height={600}
          />
        </section>
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <div className="near-places__list places__list">
            <OffersList
              placeCardsList={offersList}
              onMouseEnter={() => {}}
              onMouseLeave={() => {}}
              isAuthenticated={isAuthenticated}
              setFavorite={setFavorite}
            />
          </div>
        </section>
      </div>
    </main>
  );
};

const MemoizedOfferPropperty = memo(OfferPropperty);
export {MemoizedOfferPropperty as OfferPropperty};

