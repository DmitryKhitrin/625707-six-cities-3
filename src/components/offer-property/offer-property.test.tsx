import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";
import {Provider} from "react-redux";
import {OfferPropperty} from "./offer-property";
import {Map} from "../map/map";

import {store} from "../../mocks/mocked-store";
import {ParsedOfferCard} from "../../utils/utils";

const mocData = {
  id: `3`,
  price: 100,
  previewImage: `img/apartment-02.jpg`,
  rating: `100%`,
  type: `Private room`,
  isPremium: true,
  title: `Best place in the World really.Beautiful luxurious apartment at great location`,
  isFavorite: false,
  images: [``],
  bedrooms: 1,
  maxAdults: 1,
  goods: [``],
  host: {
    personPhoto: ``,
    personName: ``,
    id: 1,
    isPro: false
  },
  location: [52, 2],
  description: ``,
  city: {
    name: `Moscow`,
    location: [54, 2]
  }
} as unknown as ParsedOfferCard;

const offerProperties = {
  ...mocData,
  isAuthenticated: false,
  setFavorite: jest.fn(),
  reviews: [{
    comment: `Beaut`,
    date: `2020-02-14T17:36:30.592Z`,
    id: `1`,
    rating: `60%`,
    user: {
      personPhoto: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`,
      personName: `Corey`,
      id: 14,
      isPro: true,
    }
  }],
  offersList: [mocData]
};

it(`<OfferPropperty /> should render place-card.`, () => {
  Map.prototype.componentDidMount = jest.fn();
  const tree = renderer
    .create(
        <Provider store={store}>
          <Router>
            <OfferPropperty {...offerProperties} setFavorite={jest.fn()} />
          </Router>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
