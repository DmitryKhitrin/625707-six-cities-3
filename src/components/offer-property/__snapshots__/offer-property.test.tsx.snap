// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfferPropperty /> should render place-card. 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src=""
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            Best place in the World really.Beautiful luxurious apartment at great location
          </h1>
          <button
            className="property__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height={33}
              width={31}
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "100%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            4.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Private room
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            1
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            1
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            100
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height={74}
                src=""
                width={74}
              />
            </div>
            <span
              className="property__user-name"
            >
              
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              1
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height={54}
                    src="https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg"
                    width={54}
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  Corey
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  Beaut
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  January 1970
                </time>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <div
        id="map"
        style={
          Object {
            "height": 600,
          }
        }
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <div
          className="cities__places-list places__list tabs__content"
        >
          <div>
            <article
              className="cities__place-card place-card"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
            >
              <div
                className="place-card__mark"
              >
                <span>
                  Premium
                </span>
              </div>
              <div
                className="cities__image-wrapper place-card__image-wrapper"
              >
                <a
                  href="/offer/3"
                  onClick={[Function]}
                >
                  <img
                    alt="Place image"
                    className="place-card__image"
                    height="200"
                    src="img/apartment-02.jpg"
                    width="260"
                  />
                </a>
              </div>
              <div
                className="place-card__info"
              >
                <div
                  className="place-card__price-wrapper"
                >
                  <div
                    className="place-card__price"
                  >
                    <b
                      className="place-card__price-value"
                    >
                      €
                      100
                    </b>
                    <span
                      className="place-card__price-text"
                    >
                      / night
                    </span>
                  </div>
                  <button
                    className="place-card__bookmark-button button"
                    onClick={[Function]}
                    type="button"
                  >
                    <svg
                      className="place-card__bookmark-icon"
                      height="19"
                      width="18"
                    >
                      <use
                        xlinkHref="#icon-bookmark"
                      />
                    </svg>
                    <span
                      className="visually-hidden"
                    >
                      To bookmarks
                    </span>
                  </button>
                </div>
                <div
                  className="place-card__rating rating"
                >
                  <div
                    className="place-card__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <h2
                  className="place-card__name"
                >
                  <a
                    href="/offer/3"
                    onClick={[Function]}
                  >
                    Best place in the World really.Beautiful luxurious apartment at great location
                  </a>
                </h2>
                <p
                  className="place-card__type"
                >
                  Private room
                </p>
              </div>
            </article>
          </div>
        </div>
      </div>
    </section>
  </div>
</main>
`;
